runtime/jcl/common/mgmtthread.c

```diff
@@ -1120,8 +1112,7 @@ handlerMonitorWaited(J9HookInterface** hook, UDATA eventNum, void* eventData, vo
  * @retval JNI_ENOMEM OutOfMemoryError is set
  * @retval JNI_ERR FindClass() or GetMethodID() failed, an exception is set
  */
-static jint
-initIDCache(JNIEnv *env)
+JNIEXPORT jint JNICALL Java_openj9_management_internal_IDCacheInitializer_initIDCache (JNIEnv *env, jclass jcls)
```

Comment:
this linkage is not happening: I get `Caused by: java.lang.UnsatisfiedLinkError: openj9/management/internal/IDCacheInitializer.initIDCache()I` from the Java caller. I have been looking at this for a long time, but not sure what is going on, so I thought I will ask here. Suggestions?

Reply:
JNIEXPORT does nothing anywhere but windows and should not be used. You need to modify the exports for both the old and new build systems (search for the name of one of the other natives to find the required files).

Formatting:
```
jint JNICALL
Java_... {
}
```

Reply:
Also, this should not return jint - if something goes wrong it should throw an exception which will cause the class initialization to fail.